components/automate-cli/cmd/chef-automate/automateDeploymentFactory.go

```diff
@@ -0,0 +1,23 @@
+package main
+
+type deployDirector struct {
+	manager deployManager
+}
+
+func newDeployDirector(m deployManager) *deployDirector {
+	return &deployDirector{
+		manager: m,
+	}
+}
+
+func (d *deployDirector) setDeployManager(m deployManager) {
+	d.manager = m
+}
+
+func (d *deployDirector) executeDeployemnt(args []string) error {
+	return d.manager.doDeployWork(args)
```

Comment:
Can we not call `manager.doDeployWorks` where `executeDeployment` is called from. As manager is already an interface var. And is available in `runDeployCmd`.    This whole layer will not be needed, as `getDeployer` is working as Factory Method.